match/lib/match/storage/s3_storage.rb

```diff
@@ -181,7 +184,7 @@ def s3_object_path(file_name)
       end
 
       def strip_s3_object_prefix(object_path)
-        object_path.gsub(/^#{s3_object_prefix}/, "")
+        object_path.delete_prefix(s3_object_prefix.to_s).delete_prefix('/')
```

Comment:
`.to_s` is probably unnecessary here, as the field is initialized using it in https://github.com/fastlane/fastlane/blob/73f09a32eda7c7cfe8a9828b0d950a7989c4c950/match/lib/match/storage/s3_storage.rb#L68C9-L68C26
But doesn't break anything leaving it.

Reply:
iinm it fixes the case of it being potentially `nil` (which is actually the most common use case as I'd assume not many people use a custom key prefix for their _match_ S3 storage), as `.delete_prefix(nil)` would crash otherwise while `delete_prefix(nil.to_s)` won't and will just be a no-op 😉